Products/GenericSetup/registry.py

```diff
@@ -644,6 +644,10 @@ def __init__(self):
     def getProfileInfo(self, profile_id, for_=None):
         """ See IProfileRegistry.
         """
+        if profile_id.startswith("profile-"):
+            profile_id = profile_id[len('profile-'):]
```

Comment:
The point was that x[8:] is more fragile than x[len(somtheing):] So snapshot- and profile- don't get muddled by accident. I've not reviewed GenericSetup before so I'm bound to pick up on stuff.